controllers/btpoperator_controller.go

```diff
@@ -1131,3 +1179,567 @@ func (r *BtpOperatorReconciler) watchConfigPredicates() predicate.Funcs {
+	}
+	logger.Info("generation of signed webhook certificate success")
+	return nil
+}
+
+func (r *BtpOperatorReconciler) generateSignedCert(ctx context.Context, expiration time.Time, caCertificate, caPrivateKey []byte) ([]byte, []byte, error) {
+	if caCertificate == nil || caPrivateKey == nil {
+		data, err := r.getDataFromSecret(ctx, CaSecret)
+		if err != nil {
+			return []byte{}, nil, err
+		}
+
+		caCertificate, err = r.getValueByKey(r.buildKeyNameWithExtension(CaSecretDataPrefix, CertificatePostfix), data)
+		if err != nil {
+			return []byte{}, nil, err
+		}
+
+		caPrivateKey, err = r.getValueByKey(r.buildKeyNameWithExtension(CaSecretDataPrefix, RsaKeyPostfix), data)
+		if err != nil {
+			return []byte{}, nil, err
+		}
+	}
+
+	webhookCertificate, webhookPrivateKey, err := certs.GenerateSignedCertificate(expiration, caCertificate, caPrivateKey)
+	if err != nil {
+		return []byte{}, nil, err
+	}
+
+	return webhookCertificate, webhookPrivateKey, err
+}
+
+func (r *BtpOperatorReconciler) appendCertificationDataToUnstructured(us *[]*unstructured.Unstructured, certName string, certificate, privateKey []byte, prefix string) error {
+	err, data := r.mapCertToSecretData(certificate, privateKey, r.buildKeyNameWithExtension(prefix, CertificatePostfix), r.buildKeyNameWithExtension(prefix, RsaKeyPostfix))
+	if err != nil {
+		return err
+	}
+	secret := r.buildSecretWithDataAndLabels(certName, data, map[string]string{managedByLabelKey: operatorName})
+
+	unstructuredObj, err := runtime.DefaultUnstructuredConverter.ToUnstructured(secret)
+	if err != nil {
+		return err
+	}
+	*us = append(*us, &unstructured.Unstructured{Object: unstructuredObj})
+	return nil
+}
+
+func (r *BtpOperatorReconciler) mapCertToSecretData(certificate, privateKey []byte, keyNameForCert, keyNameForPrivateKey string) (error, map[string][]byte) {
+	return nil, map[string][]byte{
+		keyNameForCert:       certificate,
+		keyNameForPrivateKey: privateKey,
+	}
+}
+
+func (r *BtpOperatorReconciler) reconcileWebhooksConfigurations(ctx context.Context, resourcesToApply *[]*unstructured.Unstructured, expectedCa []byte) error {
+	logger := log.FromContext(ctx)
+	logger.Info("starting reconcilation of webhooks")
+
+	if expectedCa == nil {
+		secret := &corev1.Secret{}
+		if err := r.Get(ctx, client.ObjectKey{Namespace: ChartNamespace, Name: CaSecret}, secret); err != nil {
+			return err
+		}
+		ca, ok := secret.Data[r.buildKeyNameWithExtension(CaSecretDataPrefix, CertificatePostfix)]
+		if !ok || ca == nil {
+			return fmt.Errorf("while receiving certificate data from CA secret in reconcilation webhook")
+		}
+		expectedCa = ca
+	}
+
+	for i, resource := range *resourcesToApply {
+		switch kind := resource.GetKind(); kind {
+		case MutatingWebhookConfiguration, ValidatingWebhookConfiguration:
+			{
+				err := r.reconcileWebhook(ctx, &resource, expectedCa)
+				if err != nil {
+					return err
+				}
+				(*resourcesToApply)[i] = resource
```

Comment:
`reconcileWebhook` mutates the webook pointed by `resource` var, the changes should be already reflected in the slice without the need to overwrite the element in the slice